bugbug/repository.py

```diff
@@ -241,24 +241,47 @@ def _transform(commit):
             type_ = "Rust"
         else:
             type_ = ext
-        commit.types.add(type_)
 
+        size = None
         if not stats["deleted"]:
             try:
                 after = HG.cat([path.encode("utf-8")], rev=commit.node.encode("ascii"))
-                sizes.append(after.count(b"\n"))
+                size = after.count(b"\n")
             except hglib.error.CommandError as e:
                 if b"no such file in rev" not in e.err:
                     raise
 
+        if is_test(path):
+            commit.test_files_modified_num += 1
+
+            commit.test_added += stats["added_lines"]
+            commit.test_deleted += stats["deleted_lines"]
+
+            if size is not None:
+                test_sizes.append(size)
+        else:
+            commit.files_modified_num += 1
+
+            commit.added += stats["added_lines"]
+            commit.deleted += stats["deleted_lines"]
+
+            if size is not None:
+                sizes.append(size)
+            commit.types.add(type_)
+
     commit.total_file_size = sum(sizes)
     commit.average_file_size = (
         commit.total_file_size / len(sizes) if len(sizes) > 0 else 0
     )
     commit.maximum_file_size = max(sizes, default=0)
     commit.minimum_file_size = min(sizes, default=0)
 
-    commit.files_modified_num = len(patch_data)
+    commit.total_test_file_size = sum(test_sizes)
+    commit.average_test_file_size = (
+        commit.total_test_file_size / len(test_sizes) if len(test_sizes) > 0 else 0
```

Comment:
nit: we should have a function for that stuff since we've this pattern several times